.github/workflows/release.yml

```diff
@@ -0,0 +1,41 @@
+# This is a GitHub workflow to auto-deploy releases on tags
+# Useful references:
+# https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+# https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-python#publishing-to-package-registries
+# https://github.community/t/how-to-run-github-actions-workflow-only-for-new-tags/16075
+# https://www.codingwithcalvin.net/git-tag-based-released-process-using-github-actions/
+name: Release
+
+# Trigger the workflows on all PRs pushed commits and/or tags
+# to be able to test changes without opening PRs
+on:
+  push:
+    # FIXME: use the release event instead?
+    tags: [0-9]+.[0-9]+.[0-9]+
+    branches: master
+
+jobs:
+  build-n-publish:
+    name: Build and publish on tags to PyPI
```

Comment:
It will always build, but only also do the publish step on (specific) tags.

@minrk did the same in https://github.com/jupyterhub/oauthenticator/pull/383. If possible I'd like to get the "dummy publish" to run as a test under a workflow not named Release/Publish or similarly.

I think it will avoid reactions like "Oh did my push to master trigger a release!?!", to then be followed by mandatory precautionary inspection leading to "oh, the specific step to push wasn't triggered".

Also, if someone then push a tag that didn't trigger the workflow (1.0.0b1 for example), but it triggered by a normal push of a commit to the main branch, then that user would see a release/publish workflow be triggered and succeeding (but as master), so that user would in this case think it was all alright, but then not see any PyPI package because it never ran on the tagged commit, only on the master.

And perhaps finally, a user that wants to make use of a development release or similarly may inspect and see a Publish workflow and assume there is a development release on pushes to master or similarly.

## Summary
I consider the value of always testing the build of the package, and personally think it adds confusion to do so but only conditionally publish to PyPi under something named Publish or Release.